spec/controllers/works/default_rails_actions_spec.rb

```diff
@@ -176,10 +176,43 @@ def call_with_params(params)
     end
   end
 
+  describe "edit" do
+    let(:user) { create(:user) }
+    let(:work) { create(:work, authors: [user.default_pseud], posted: true) }
+
+    before do
+      fake_login_known_user(user)
+    end
+
+    it "redirects to orphan work page if only author is being removed" do
+      get :edit, params: { id: work.id, remove: "me" }
+      expect(response).to redirect_to controller: "orphans", action: "new", work_id: work.id
+    end
+  end
+
+  context "destroy" do
+    let(:user) { create(:user) }
+    let!(:work) { create(:work, authors: [user.default_pseud], posted: true) }
+
+    before do
+      fake_login_known_user(user)
+    end
+
+    it "sets flash message in case of error" do
+      allow_any_instance_of(Work).to receive(:destroy).and_raise("Cannot save")
+
+      delete :destroy, params: { id: work }
+      expect(flash[:error]).to eq("We couldn't delete that right now, sorry! Please try again later.")
+
+      allow_any_instance_of(Work).to receive(:destroy).and_call_original
```

Comment:
I feel like RSpec ought to be able to clean up its own mocks. Have you checked whether this line of code is necessary?

Reply:
Looks like it works fine without